kartothek/io/dask/dataframe.py

```diff
@@ -301,3 +304,66 @@ def update_dataset_from_ddf(
         metadata=metadata,
         metadata_merger=metadata_merger,
     )
+
+
+def collect_dataset_statistics(
+    store, dataset_uuid, table_name, predicates=None, frac=0.3, factory=None,
+):
+    """
+    Retrieve some statistics, maybe even plot them, generate a report...?
+
+    Parameters
+    ----------
+    store
+      A factory function providing a KeyValueStore
+    dataset_uuid
+      The dataset's unique identifier
+    table_name
+      Name of the kartothek table for which to retrieve the statistics
+    predicates
+      Kartothek predicates to apply filters on the data for which to gather statistics
+
+      .. warning::
+          Filtering will only be applied for predicates on indices.
+          The evaluation of the predicates therefore will therefore only return an approximate result.
+
+    frac
+      Fraction of the data to use for gathering statistics. `frac == 1` will use the whole dataset.
+    factory
+       A DatasetFactory holding the store and UUID to the source dataset.
+
+    Returns
+    -------
+    A dask.dataframe containing information about dataset statistics.
+
+    Raises
+    ------
+    ValueError
+      If no metadata could be retrieved, raise an error.
+
+    """
+    dataset_factory = _ensure_factory(
+        dataset_uuid=dataset_uuid,
+        store=store,
+        factory=factory,
+        load_dataset_metadata=False,
+    )
+    mp_iterator = random_sample(
+        frac,
+        dispatch_metapartitions_from_factory(dataset_factory, predicates=predicates),
+    )
+    dfs = [
+        dask.delayed(MetaPartition.get_parquet_metadata)(
+            mp, store=dataset_factory.store_factory, table_name=table_name
+        )
+        for mp in mp_iterator
+    ]
+    if not dfs:
+        raise ValueError(
+            f"Could not retrieve metadata for dataset {dataset_uuid}. Perhaps the value {frac} for the parameter `frac` is too small?"
```

Comment:
Or the dataset is empty? I think guesses in the exception message are not a really good UX.

Reply:
I could make a point that `frac>0` should never return an empty value, i.e. we should always "round up" / ensure there is always at least one partition. That's probably not that easy with the itertoolz sampler but would also not be overly complicated to achieve. (One could make the opposite argument as well)

Opinions? Suggestions? As long as the behaviour is not surprising / documented, I'm fine either way

Reply:
I agree, returning empty data here feels weird.
I guess the intention was saying something like "Hey user, your choice for `frac` is really off"...

Reply:
So even if we agree on rounding up and rejecting `frac == 0`, `dfs` might still be empty in case of an empty dataset. And an empty dataset is valid data. It's an edge case for sure but asking the user to guard each and every call to this function with `if len(ds.partitions) == 0` is not a very usable API and I think we'll run into bugs all over the place (even in our own code) when we do that.

So what I would rather propose:

- raise if `frac` is not in `(0, 1]` (half closed interval)
- drop the `if not dfs` check and write a test case for an empty DS. You'll likely need to tell dask the dtype for each and every column (because it cannot infer it, which is suboptimal anyways and requires it to eagerly load+test the first partition)

Reply:
+1 for checking the input for`frac`.

Just to make sure, I understood your proposal: You are in favor of *not* raising for an empty dataset, but instead returning an empty result df and maybe issue a warning?

Reply:
> Just to make sure, I understood your proposal: You are in favor of not raising for an empty dataset, but instead returning an empty result df and maybe issue a warning?

yes.

Reply:
Should be fixed: sampling is now done differently and rounding up to at least one metapartition is guarateed.
Renamed to `collect_metadata` and no normalization.

- [ ] Is there a test for the rounding up?